libzeth/tests/core/field_element_utils_test.cpp

```diff
@@ -35,20 +37,35 @@ TEST(FieldElementUtilsTest, BigIntEncodeDecode)
     ASSERT_EQ(bi, bi_decoded);
 }
 
-TEST(FieldElementUtilsTest, FieldElementEncodeDecode)
+TEST(FieldElementUtilsTest, BaseFieldElementEncodeDecode)
 {
     Fr fe = dummy_field_element();
-    std::string fe_hex = libzeth::field_element_to_hex<Fr>(fe);
+    std::string fe_hex = libzeth::base_field_element_to_hex<Fr>(fe);
     std::cout << "fe_hex: " << fe_hex << std::endl;
-    Fr fe_decoded = libzeth::field_element_from_hex<Fr>(fe_hex);
+    Fr fe_decoded = libzeth::base_field_element_from_hex<Fr>(fe_hex);
+    ASSERT_EQ(fe, fe_decoded);
+}
+
+TEST(FieldElementUtilsTest, ExtFieldElementEncodeDecode)
```

Comment:
Yes definitely, but weren't you working toward removing any "default" template instantiation from `libzeth` in order to aim toward a fully generic library and generic tests (executed - as in libff on various curves)? This seems like something to carry out on a separate PR as it is of broader interest

Reply:
Yeah, I guess it could be done independently.  I mention it here because it's related to testing that this new generic code (i.e. the library code has become more generic, particularly if we support `to_json` over all field types, so it would be nice to also make the tests generic), but it could conflict with the PR to remove the default.
(To be clear, it could conflict in terms of merging, but is consistent in the sense of removing hard-coded types in the library ... if that makes sense).
In particular, doing this for G2 proto utils tests could clear up the issue with bw6-761 if that is actually an issue.

Reply:
> (To be clear, it could conflict in terms of merging, but is consistent in the sense of removing hard-coded types in the library ... if that makes sense)

Yes definitely in line with the other PR. Ok, I'll have a look into this and make the test generic on my end then :)